extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -177,29 +242,27 @@ class CornerstoneViewportService implements IViewportService {
     viewportIndex: number,
     viewportData: StackViewportData | VolumeViewportData,
     publicViewportOptions: PublicViewportOptions,
-    publicDisplaySetOptions: DisplaySetOptions[]
+    publicDisplaySetOptions: DisplaySetOptions[],
+    presentation?: Presentation
   ): void {
     const renderingEngine = this.getRenderingEngine();
-    const viewportInfo = this.viewportsInfo.get(viewportIndex);
-
-    if (!publicViewportOptions.viewportId) {
-      publicViewportOptions.viewportId = this.getViewportId(viewportIndex);
+    const viewportId =
+      publicViewportOptions.viewportId || this.getViewportId(viewportIndex);
+    if (!viewportId) {
```

Comment:
shouldn't `publicViewportOptions` be updated with the new viewportId? I guess so, since we were doing it before as well

Reply:
No, you can't set the viewportId in the public viewport options - this is an internal/non-public detail.